send-transaction-service/src/transaction_client.rs

```diff
@@ -285,6 +288,60 @@ where
         lock.1.cancel();
         Ok(())
     }
+
+    async fn do_update_key(&self, identity: &Keypair) -> Result<(), Box<dyn std::error::Error>> {
+        let runtime_handle = self.runtime_handle.clone();
+        let config = Self::create_config(
+            Some(identity.insecure_clone()),
+            self.leader_forward_count as usize,
+        );
+        let leader_updater = self.leader_updater.clone();
+        let handle = self.join_and_cancel.clone();
+
+        let join_handle = {
+            let Ok(mut lock) = handle.lock() else {
+                return Err("TpuClientNext task panicked.".into());
+            };
+            lock.1.cancel();
+            lock.0.take() // Take the `join_handle` out of the critical section
+        };
+
+        if let Some(join_handle) = join_handle {
+            let Ok(result) = join_handle.await else {
```

Comment:
nit: you don't need let else ... and then a match below. You can do everything
in the match.

Reply:
I try to use early-exit here to avoid deep indentation. With match I will have to do 

```rust
match join_handle.await {
   Ok(result) => {
          match result {
              Ok(_,_) => ...,
              Err(e) => ...
          }
   }
   Err(_) =>  {}

}
```